src/motor_control_circ/src/MoveMotor.cpp

```diff
@@ -0,0 +1,37 @@
+#include <ros/ros.h>
+#include <geometry_msgs/Twist.h>
+
+#include <MoveMotor.h>
+#include <phidget22.h>
+#include <unistd.h>
+#include <string>
+
+MoveMotor::MoveMotor(int argc, char **argv, std::string node_name) {
+    ros::init(argc, argv, node_name);
+    ros::NodeHandle nh;
+    ros::NodeHandle private_nh("~");
+
+    std::string subscribe_topic = "subscribe_topic";
+    int queue_size = 10;
+    my_subscriber = nh.subscribe(subscribe_topic, queue_size, &MoveMotor::callback, this);
+}
+
+void MoveMotor::callback(const geometry_msgs::Twist::ConstPtr& msg) {
+    ROS_INFO("Received Twist");
+
+    PhidgetBLDCMotorHandle bldcMotor0;
+
+    PhidgetBLDCMotor_create(&bldcMotor0);
+
+    int channel = 2;
+    Phidget_setHubPort((PhidgetHandle)bldcMotor0, channel);
+
+    Phidget_openWaitForAttachment((PhidgetHandle)bldcMotor0, 5000);
```

Comment:
what happens if the motor is not detected for a long enough time?

Reply:
I believe that's only the case when the motor is not actually connected, i.e. there's nothing to attach to.